drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -322,7 +424,8 @@ static void siwx917_iface_init(struct net_if *iface)
 	sl_wifi_set_scan_callback(siwx917_on_scan, sidev);
 	sl_wifi_set_join_callback(siwx917_on_join, sidev);
 
-	status = sl_wifi_get_mac_address(SL_WIFI_CLIENT_INTERFACE, &sidev->macaddr);
+	sidev->interface = sl_wifi_get_default_interface();
+	status = sl_wifi_get_mac_address((sidev->interface & GET_INTERFACE), &sidev->macaddr);
```

Comment:
I suggest `FIELD_GET(GET_INTERFACE, sidev->interface)`. Also `GET_INTERFACE` is not well chosen. I should be prefixed by `SIWX91X`, so we know this value does not come zephyr of wiseconnect. Then, it should contains `MASK` rather than `GET`.

In addition, you mean the value returned `sl_wifi_get_default_interface()` can't be used as-is?